### PR TITLE
Network switcher

### LINKED ISSUE #93: Add network switcher
Currently, the network is shown only. No network management yet.

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds a network switcher when clicking on the network name
* fixes #93 
* uses mostly code from explorer
* introduces `NetworkInstance` where pool contracts are expected. Currently, it is mainnet, testnet and devnet.

image

### CODE REVIEW

**Comment by markmhx** [comment]
@friedger mind providing a screen recording of the interactions?

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "@emotion/react": "11.10.6",
     "@emotion/styled": "11.10.6",
     "@noble/hashes": "1.3.0",
+    "@reduxjs/toolkit": "^1.9.3",
```
Comment (kyranjamie) [inline package.json:21]: Use the `-E` flag when installing packages to pin to a version

File: src/utils/networks.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export function createNewTo(to: To, network: Network): To {
+  // if "to" is a string must not contain a search query
+  // it is overwritten anyway
+  const pathname = typeof to === 'string' ? to : to.pathname;
+  return {
+    pathname,
+    search: createSearch(network),
+  };
+}
```
Comment (kyranjamie) [inline src/utils/networks.ts:17]: Don't really understand what's happening here?

File: src/pages/settings/network/network.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <BaseDrawer title={'Select network'} isShowing onClose={() => navigate(routes.HOME)}>
+      <Flex alignItems="center" flexDirection="column" pb={['loose', '48px']} px="loose">
+        <InfoCard width="420px">
+          <NetworkItems />
+        </InfoCard>
+      </Flex>
+    </BaseDrawer>
+  );
+};
```
Comment (kyranjamie) [inline src/pages/settings/network/network.tsx:23]: Elsewhere we [avoid FC](https://medium.com/raccoons-group/why-you-probably-shouldnt-use-react-fc-to-type-your-react-components-37ca1243dd13) and favour defining components as function definitions

File: src/pages/settings/network/network-items.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+const Item: React.FC<ItemProps> = ({ item, isActive, isDisabled, onClick, isCustom, ...rest }) => {
+  const {
+    removeCustomNetwork,
+    apiUrls: { mainnet, testnet },
+  } = useGlobalContext();
+  const isMainnet = item.url === mainnet;
+  const isTestnet = item.url === testnet;
+  const isDefault = isMainnet || isTestnet;
+  let itemNetworkId: ChainID.Mainnet | ChainID.Testnet = isMainnet
+    ? ChainID.Mainnet
```
Comment (kyranjamie) [inline src/pages/settings/network/network-items.tsx:81]: We have a `whenChainId` function you could copy from the wallet repo

Comment (friedger) [reply src/pages/settings/network/network-items.tsx:81]: @kyranjamie Not sure how to apply this here. I applied it to network mode

File: src/pages/settings/network/network.tsx (hunk 3 above)
Comment (friedger) [reply src/pages/settings/network/network.tsx]: :+1:

File: src/utils/networks.ts (hunk 2 above)
Comment (friedger) [reply src/utils/networks.ts]: I renamed the function and added a comment. 
The function replaces the search

File: package.json (hunk 1 above)
Comment (friedger) [reply package.json]: :+1:

**Review by kyranjamie — APPROVED** [review]
Thanks Friedger, please rebase your commits into one or two before merging

**Comment by friedger** [comment]
@kyranjamie Any ideas why these checks fail?

**Comment by friedger** [comment]
Merged by #115

**Pull request closed without merging**